edb/schema/reflection/writer.py

```diff
@@ -956,9 +956,27 @@ def write_meta_delete_object(
 
             blocks.append((parent_update_query, parent_variables))
 
+        # We need to delete any links create via reflection_proxy
+        layout = classlayout[mcls]
+        proxy_links = [
+            link for link, layout_entry in layout.items()
+            if layout_entry.reflection_proxy
+        ]
+
+        operations = []
+        if proxy_links:
+            # Link deletion triggers apparently don't work for the
```

Comment:
It's not that triggers don't fire: we don't _have_ a forward cascade logic at all (i.e. `ON SOURCE DELETE DELETE TARGET`).

Reply:
I think you missed my meaning here: we *do* have triggers for deleting entries from the link table when the source is deleted (but not for deleting the target object, like you said), and that's not working here, which results in leaving stray dangling entries in the link table).

Reply:
Oh, that is unexpected.  There is no exception for system tables, they're treated exactly like user schema, so we might be papering over a bug here.

Reply:
Yeah, you are right, I was sloppy about this.

I dug into it and there are actually a couple of bugs involving inheritance and trigger tables.